controllers/handlers_utils.go

```diff
@@ -211,21 +215,30 @@ func updateResource(ctx context.Context, dr dynamic.ResourceInterface,
 	}
 
 	l := logger.WithValues("resourceNamespace", object.GetNamespace(),
-		"resourceName", object.GetName(), "resourceGVK", object.GetObjectKind().GroupVersionKind())
+		"resourceName", object.GetName(), "resourceGVK", object.GetObjectKind().GroupVersionKind(), "subresources", subresources)
 	l.V(logs.LogDebug).Info("deploying policy")
 
 	data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, object)
 	if err != nil {
 		return err
 	}
 
+	_, err = patchRessource(ctx, dr, object.GetName(), data, subresources, l)
+	return err
+}
+
+func patchRessource(ctx context.Context, dr dynamic.ResourceInterface,
+	name string, data []byte, subresources []string, logger logr.Logger) (*unstructured.Unstructured, error) {
+
+	l := logger.WithValues("subresources", subresources)
+	l.V(logs.LogDebug).Info("patching")
+
 	forceConflict := true
 	options := metav1.PatchOptions{
 		FieldManager: "application/apply-patch",
 		Force:        &forceConflict,
 	}
-	_, err = dr.Patch(ctx, object.GetName(), types.ApplyPatchType, data, options)
-	return err
+	return dr.Patch(ctx, name, types.ApplyPatchType, data, options, subresources...)
```

Comment:
just for my understanding. If we pass sub resources to Patch, it will Path both Spec and Status (assuming Status is the sub resource we want to Patch)?

Reply:
No currently if you set subressources it will only patch the subresource and ignore the other stuff. I tried by patching two times once with subresources and one without but then you have to update the dynamic ressourceinterface since a changed happend => "Operation cannot be fulfilled on services \"service0\": the object has been modified; please apply your changes to the latest version and try again"

Reply:
Thanks. So you tried 

```
_, err = dr.Patch(ctx, object.GetName(), types.ApplyPatchType, data, options) // no sub resources 
if err != nil {
  return err
}

return dr.Patch(ctx, name, types.ApplyPatchType, data, options, subresources...)
```

and that failed? 

I will have to look at dynamic resource interface. Cause, the object is only used when we do Patch. So I thought above would have worked.